src/pages/ContentScripts/features/fast-pr/index.tsx

```diff
@@ -55,22 +60,48 @@ const init = async (matchedUrl: MatchedUrl | null) => {
     document.body.appendChild(container);
   }
 };
+
+const checkCacheAndInit = (url: string) => {
+  const cachedData = localStorage.getItem(CACHE_KEY);
+  const currentTime = Date.now();
+
+  if (cachedData) {
+    const { matchedFun, timestamp } = JSON.parse(cachedData);
+    if (currentTime - timestamp < CACHE_EXPIRY) {
+      const iframeElement = document.getElementById('sandboxFrame') as HTMLIFrameElement;
+      if (iframeElement && iframeElement.contentWindow) {
+        iframeElement.contentWindow.postMessage({ command: 'useCachedData', matchedFun: matchedFun, url: url }, '*');
+      }
+    }
+    return;
+  }
+
+  const iframeElement = document.getElementById('sandboxFrame') as HTMLIFrameElement;
+  if (iframeElement && iframeElement.contentWindow) {
+    iframeElement.contentWindow.postMessage({ command: 'requestMatchedUrl', matchedFun: null, url: url }, '*');
+  }
+};
+
 chrome.runtime.onMessage.addListener((message) => {
   if (message.type === 'urlChanged') {
     handleUrlChange(message.url);
   }
 });
+
 function handleUrlChange(url: string) {
   const existingContainer = document.getElementById(featureId);
   if (existingContainer) {
     existingContainer.remove();
   }
-  const iframe = document.getElementById('sandboxFrame') as HTMLIFrameElement;
-  if (iframe && iframe.contentWindow) {
-    iframe.contentWindow.postMessage({ command: 'matchUrl', url: url }, '*');
-  }
+  checkCacheAndInit(url);
 }
+
 window.addEventListener('message', (event: MessageEvent) => {
+  if (event.data && event.data.matchedFun) {
+    const matchedFun = event.data.matchedFun;
+    const currentTime = Date.now();
+    localStorage.setItem(CACHE_KEY, JSON.stringify({ matchedFun, timestamp: currentTime }));
```

Comment:
Why do you set the `timestamp` to current time for both scenarios? I think we only need to update the cache if the `matchedFunc` is returned with newly fetch.

If so, the config will not be updated if a user keep using the cache without a gap more than 1 hour.

Reply:
Modified and improved.